sdk/src/main/java/com/sygic/travel/sdk/provider/DataProvider.kt

```diff
@@ -161,7 +161,7 @@ internal class DataProvider(
 	fun addPlaceToFavorites(id: String, back: Callback<String>?) {
 		Thread(Runnable {
 			val insertedCount = stDb?.favoriteDao()?.insert(Favorite(id))
-			if (insertedCount == 1L) {
+			if (insertedCount != null && insertedCount > 0L) {
```

Comment:
The thing is that the method doesn't return a number of inserted rows, but a `rowid` of an inserted row.